examples/echo_notification.rs

```diff
@@ -38,19 +40,22 @@ use std::time::Duration;
 
 /// event loop for the client
 async fn client_event_loop(mut litep2p: Litep2p, mut handle: NotificationHandle, peer: PeerId) {
-    // open substream to `litep2p` and since no connection exists, dial the peer first
+    // open substream to `peer`
+    //
+    // if `litep2p` is not connected to `peer` but it has at least one known address,
+    // `NotifcationHandle::open_substream()` will automatically dial `peer`
     handle.open_substream(peer).await.unwrap();
 
     // wait until the substream is opened
-    let peer = loop {
+    loop {
         tokio::select! {
             _ = litep2p.next_event() => {}
             event = handle.next() => match event.unwrap() {
-                NotificationEvent::NotificationStreamOpened { peer, .. } => break peer,
+                NotificationEvent::NotificationStreamOpened { .. } => break,
                 _ => {},
             }
         }
-    };
+    }
```

Comment:
Just to make sure, that we are using `vec![1, 3, 3, 7]` both for handshake and for actual message doesn't mean that we must send the handshake manually?

Reply:
Yeah, it's just lack of imagination that both the notification that it sends and the handshake it uses are the same. But handshake is sent automatically by `NotificationProtocol`